### PR TITLE
Fixes #52: Add old-new parser migration guides

### LINKED ISSUE #52: Migration guide from raml-java-parser
Due on  [raml-java-parser](https://github.com/raml-org/raml-java-parser) DEPRECATION NOTICE migration guide is required.

And I think I am posting issue in right repo because 

- Due raml-java-parser inactivity it is very hard to get answer [the same question](https://github.com/raml-org/raml-java-parser/issues/672)
- Both projects belongs to the same organization raml.org and organizations which uses this libraries (smartbear & co) should not write this guides and does not provide support for libraries they don't own.

### PR DESCRIPTION
**Pull request by postatum**
Fixes #52 

After this PR is merged we should regenerate gh-pages docs with:
```sh
./scripts/generate-gh-pages-docs.sh
```

### CODE REVIEW

File: docs/migration-guide-java.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+---
+---
+
+# Migration guide (Java)
+Welcome!
+
+As you may already know RAML 0.8/1.0 Java parser `raml-java-parser` has been deprecated in favor of the new and better one - `webapi-parser`. This guide describes how to migrate an existing code from `raml-java-parser` to `webapi-parser`.
```
Comment (jstoiko) [inline docs/migration-guide-java.md]: > in favor of the new and better one

let's avoid qualitative adjectives. we can simply say "in favor of `webapi-parser`"

File: docs/migration-guide-js.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+---
+---
+
+# Migration guide (JS)
+Welcome!
+
+As you may already know RAML 0.8/1.0 JS parser `raml-1-parser` has been deprecated in favor of the new and better one - `webapi-parser`. This guide describes how to migrate an existing code from `raml-1-parser` to `webapi-parser`.
```
Comment (jstoiko) [inline docs/migration-guide-js.md]: > in favor of the new and better one

let's avoid qualitative adjectives. we can simply say "in favor of `webapi-parser`"

File: docs/migration-guide-java.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+## Detailed migration examples
+This section lists migration examples of the most common `raml-java-parser` parsing and model methods. Snippets are separated with a newline. First line of each example shows `raml-java-parser` method usage, while the second line shows how to achieve the same functionality with `webapi-parser` if possible. If no obvious alternative exists, a comment gives more detail.
+
+### Parsers
+```java
+import org.raml.v2.api.RamlModelBuilder;
+import webapi.Raml10;
+
+new RamlModelBuilder().buildApi(ramlFile);
```
Comment (jstoiko) [inline docs/migration-guide-java.md]: what does it do? I think we need to add a short description as comment before each block below

File: docs/migration-guide-java.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+Raml08.parse(content, ramlLocation).get();
+```
+
+### API Models
+```java
+import org.raml.v2.api.RamlModelResult;
+import org.raml.v2.api.model.v10.api.Api;
+import webapi;
+
+Api oldModel = new RamlModelBuilder().buildApi(input).getApiV08();
```
Comment (jstoiko) [inline docs/migration-guide-java.md]: add a short description before each block

File: docs/migration-guide-js.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+const oldParser = require('raml-1-parser')
+const wp = require('webapi-parser')
+const wap = wp.WebApiParser
+
+const node = await oldParser.loadApi('/path/to/api.raml')
+const model = await wap.raml10.parse('file:///path/to/api.raml')
+// or
+// const model = wap.raml08.parse('file:///path/to/api.raml')
+
+oldParser.asFragment(node)
```
Comment (jstoiko) [inline docs/migration-guide-js.md]: add a short description of what it does before each block

File: docs/migration-guide-js.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+const node = await oldParser.loadApi('/path/to/api.raml')
+const model = await wap.raml10.parse('file:///path/to/api.raml')
+// or
+// const model = wap.raml08.parse('file:///path/to/api.raml')
+
+node.resources()
+model.encodes.endPoints()
+// Note that webapi-parser resources are flat and occur in the order defined in the RAML doc.
+
+node.resources()[0].methods()
```
Comment (jstoiko) [inline docs/migration-guide-js.md]: add a short description of what it does before each block

File: docs/migration-guide-java.md (hunk 1 above)
Comment (postatum) [reply docs/migration-guide-java.md]: Fixed.

File: docs/migration-guide-js.md (hunk 2 above)
Comment (postatum) [reply docs/migration-guide-js.md]: Fixed.

File: docs/migration-guide-java.md (hunk 3 above)
Comment (postatum) [reply docs/migration-guide-java.md]: Added comments.

File: docs/migration-guide-java.md (hunk 4 above)
Comment (postatum) [reply docs/migration-guide-java.md]: Added

File: docs/migration-guide-js.md (hunk 5 above)
Comment (postatum) [reply docs/migration-guide-js.md]: Added

File: docs/migration-guide-js.md (hunk 6 above)
Comment (postatum) [reply docs/migration-guide-js.md]: Added

**Pull request merged**